android/java/org/chromium/chrome/browser/crypto_wallet/fragments/TxFragment.java

```diff
@@ -407,27 +410,65 @@ private void setupView(View view) {
                 mAssetPrices, mSolanaEstimatedTxFee, mFullTokenList, mNativeAssetsBalances,
                 mBlockchainTokensBalances);
 
+        if (mIsSolanaInstruction) {
+            TextView gasTxTv = view.findViewById(R.id.frag_tx_tv_gas);
+            TextView totalHeading = view.findViewById(R.id.frag_tx_tv_total_heading);
+
+            gasTxTv.setText(R.string.brave_wallet_allow_spend_transaction_fee);
+            totalHeading.setText(R.string.brave_wallet_confirm_transaction_amount_fee);
+            SolanaTxData solanaTxData = TransactionUtils.safeSolData(mTxInfo.txDataUnion);
+            if (solanaTxData != null && solanaTxData.sendOptions != null) {
+                SolanaSendTransactionOptions sendTxOptions = solanaTxData.sendOptions;
+                LinearLayout sendOptionsLl = view.findViewById(R.id.frag_tx_ll_send_options);
+                if (sendTxOptions.maxRetries != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
+                    tvLabel.setText(R.string.brave_wallet_solana_max_retries);
+                    tvVal.setText(String.valueOf(sendTxOptions.maxRetries.maxRetries));
+                    sendOptionsLl.addView(tvLabel);
+                    sendOptionsLl.addView(tvVal);
+                }
+                if (sendTxOptions.preflightCommitment != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
+                    tvLabel.setText(R.string.brave_wallet_solana_preflight_commitment);
+                    tvVal.setText(sendTxOptions.preflightCommitment);
+                    sendOptionsLl.addView(tvLabel);
+                    sendOptionsLl.addView(tvVal);
+                }
+                if (sendTxOptions.skipPreflight != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
+                    tvLabel.setText(R.string.brave_wallet_solana_skip_preflight);
+                    tvVal.setText(String.valueOf(sendTxOptions.skipPreflight.skipPreflight));
+                    sendOptionsLl.addView(tvLabel);
+                    sendOptionsLl.addView(tvVal);
```

Comment:
the same with
```
                    sendOptionsLl.addView(tvLabel);
                    sendOptionsLl.addView(tvVal);
```
but should be called once after all conditions.

Reply:
Same as [this](https://github.com/brave/brave-core/pull/15593#discussion_r1007700384), all three are different.